common/functions/src/scalars/strings/regexp_like.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::fmt;
+use std::sync::Arc;
+
+use bstr::ByteSlice;
+use common_datavalues::prelude::*;
+use common_datavalues::TypeID;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use itertools::izip;
+use regex::bytes::Regex as BytesRegex;
+use regex::bytes::RegexBuilder as BytesRegexBuilder;
+
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+use crate::scalars::FunctionDescription;
+
+#[derive(Clone)]
+pub struct RegexpLikeFunction {
+    display_name: String,
+}
+
+impl RegexpLikeFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(Self {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create)).features(
+            FunctionFeatures::default()
+                .deterministic()
+                .variadic_arguments(2, 3),
+        )
+    }
+}
+
+impl Function for RegexpLikeFunction {
+    fn name(&self) -> &str {
+        &self.display_name
+    }
+
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        let not_string = args.iter().any(|arg| arg.data_type_id() != TypeID::String);
+        if not_string {
+            return Err(ErrorCode::BadArguments(format!(
+                "Expected string arg, but got {:?}",
+                args,
+            )));
+        }
```

Comment:
https://github.com/datafuselabs/databend/blob/c5ef88c8de10b4b15a134c5623f5a468c70b8685/common/functions/src/scalars/function_common.rs#L19

assert_string is just right for this purpose.